src/coreclr/jit/lsra.cpp

```diff
@@ -836,6 +837,28 @@ LinearScan::LinearScan(Compiler* theCompiler)
     rbmIntCalleeTrash = compiler->rbmIntCalleeTrash;
     regIntLast        = compiler->regIntLast;
     isApxSupported    = compiler->canUseApxEncoding();
+    if (isApxSupported)
+    {
+        regIndices = theCompiler->getAllocator(CMK_LSRA).allocate<regNumber>(ACTUAL_REG_COUNT + 1);
+        for (int i = 0; i <= ACTUAL_REG_COUNT; i++)
+        {
+            regIndices[i] = static_cast<regNumber>(i);
+        }
+    }
+    else
+    {
+        regIndices =
+            new regNumber[]{(regNumber)0,  (regNumber)1,  (regNumber)2,  (regNumber)3,  (regNumber)4,  (regNumber)5,
+                            (regNumber)6,  (regNumber)7,  (regNumber)8,  (regNumber)9,  (regNumber)10, (regNumber)11,
+                            (regNumber)12, (regNumber)13, (regNumber)14, (regNumber)15, (regNumber)32, (regNumber)33,
```

Comment:
These can be `REG_RAX` and so on instead of numbers for better readability.

Reply:
Done